lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,264 @@
+    process = subprocess.Popen(
+        command.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = subprocess.check_output(
+        ("grep", "-i", common_name), stdin=process.stdout)
+    process.wait()
+    installed_flat_name = output.split('/')[0]
+    command = "{} uninstall {}".format(binary, installed_flat_name)
+    output = flatpak_command(command)
+    if 'error' in output and 'not installed' not in output:
+        return 1, output
+
+    return 0, output
+
+
+def parse_remote(remote):
+    name = remote.split('/')[-1]
+    if '.' not in name:
+        return name
+
+    return name.split('.')[0]
+
+
+def parse_flat(name):
+    if 'http://' in name or 'https://' in name:
+        app = name.split('/')[-1].split('.')[0]
+        common_name = app
+    else:
+        common_name = name
+
+    return common_name
+
+
+def add_remote(binary, remote):
+    remote_name = parse_remote(remote)
+    command = "{} remote-add --if-not-exists {} {}".format(
+        binary, remote_name, remote)
+    output = flatpak_command(command)
+    if 'error' in output:
+        return 1, output
+
+    return 0, output
+
+
+def remove_remote(binary, remote):
+    remote_name = parse_remote(remote)
+    command = "{} remote-delete --force {} ".format(binary, remote_name)
+    output = flatpak_command(command)
+    if 'error' in output and 'not found' not in output:
+        return 1, output
+
+    return 0, output
+
+
+def is_present_remote(binary, remote):
+    remote_name = parse_remote(remote).lower() + " "
+    command = "{} remote-list".format(binary)
+    output = flatpak_command(command)
+    if remote_name in output.lower():
+        return True
+
+    return False
+
+
+def is_present_flat(binary, name):
+    command = "{} list --app".format(binary)
+    flat = parse_flat(name).lower()
+    output = flatpak_command(command)
+    if flat in output.lower():
+        return True
+
+    return False
+
+
+def flatpak_command(command):
+    process = subprocess.Popen(
+        command.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = process.communicate()[0]
+
+    return output
+
+
+def main():
+    # This module supports check mode
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=False, default=''),
+            remote=dict(required=False, default=''),
+            state=dict(
+                required=False, default="present", choices=['present', 'absent'])
+        ),
+        supports_check_mode=True
+    )
+    params = module.params
+    name = params['name']
+    remote = params['remote']
+    state = params['state']
+    module_changed = False
+    location = module.get_bin_path('flatpak')
+    if location is None:
+        module.fail_json(
+            msg="cannot find 'flatpak' binary. Aborting.")
+
+    if state == 'present':
+        if remote != '' and not is_present_remote(location, remote):
+            if module.check_mode:
```

Comment:
So the test for **check_mode** should move higher up.
You can move it into **add_remote** and do it e.g. after parse_remote().